src/ucp/api/ucp.h

```diff
@@ -731,7 +731,10 @@ typedef enum {
     UCP_OP_ATTR_FIELD_RECV_INFO     = UCS_BIT(7),  /**< recv_info field */
     UCP_OP_ATTR_FIELD_MEMH          = UCS_BIT(8),  /**< memory handle field */
 
-    UCP_OP_ATTR_FLAG_NO_IMM_CMPL    = UCS_BIT(16), /**< deny immediate completion */
+    UCP_OP_ATTR_FLAG_NO_IMM_CMPL    = UCS_BIT(16), /**< deny immediate completion,
+                                                        avoid returning NULL.
+                                                        Note: don’t avoid immediate
+                                                        calling cb before returning. */
```

Comment:
@yosefe 

> " Deny immediate completion, i.e NULL cannot be returned. If a completion callback is provided, it can be called before the function returns. "
> 
> @tonycurtis WDYT?

done.